src/main/java/Duke.java

```diff
@@ -1,8 +1,51 @@
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+
 public class Duke {
-    public static void main(String[] args) {
-        String logo = "Hello! I'm Yube \n"
-                + "What can I do for you? \n" +
-                "Bye. Hope to see you again soon!";
-        System.out.println(logo);
+    private static final String LINE = "___________________________________________________________\n";
+
+    public static void main(String[] args) throws IOException {
+        BufferedReader reader = new BufferedReader(new InputStreamReader(System.in));
+        boolean hasEnded = false;
+        String botName = "Yube";
+
+        greet(botName);
+        while (!(hasEnded)) {
+            String input = reader.readLine();
+            if (input.equals("bye")) {
+                bye();
+                hasEnded = true;
+            } else {
+                repeatFunction(input);
+            }
+        }
+    }
+
+    /**
+     * Displays a greeting message
+     * 
+     * @param botName Name of the bot
+     */
+    public static void greet(String botName) {
```

Comment:
Not sure there's a need to make the botName a parameter since it's probably only going to be used during greeting. Maybe just hardcode the botName in the greeting message? Then there would be no need to declare the botName variable in main().